nestkernel/connection_manager.cpp

```diff
@@ -1218,188 +1218,202 @@ nest::ConnectionManager::split_to_neuron_device_vectors_( const size_t tid,
 }
 
 void
-nest::ConnectionManager::get_connections( std::deque< ConnectionID >& connectome,
+nest::ConnectionManager::get_connections_( const size_t tid,
+  std::deque< ConnectionID >& connectome,
   NodeCollectionPTR source,
   NodeCollectionPTR target,
   synindex syn_id,
   long synapse_label ) const
 {
-  if ( is_source_table_cleared() )
+  std::deque< ConnectionID > conns_in_thread;
+
+  ConnectorBase* connections = connections_[ tid ][ syn_id ];
+  if ( connections )
   {
-    throw KernelException(
-      "Invalid attempt to access connection information: source table was "
-      "cleared." );
+    // Passing target_node_id = 0 ignores target_node_id while getting connections.
+    const size_t num_connections_in_thread = connections->size();
+    for ( size_t lcid = 0; lcid < num_connections_in_thread; ++lcid )
+    {
+      const size_t source_node_id = source_table_.get_node_id( tid, syn_id, lcid );
+      connections->get_connection( source_node_id, 0, tid, lcid, synapse_label, conns_in_thread );
+    }
   }
 
-  const size_t num_connections = get_num_connections( syn_id );
+  target_table_devices_.get_connections( 0, 0, tid, syn_id, synapse_label, conns_in_thread );
 
-  if ( num_connections == 0 )
+  if ( conns_in_thread.size() > 0 )
```

Comment:
This part is in all three `get_connections*_()` methods. I'd suggest to move it to the outer method and then define `conns_in_threads` in that method and pass it into the "submethods" instead of `connectome`. That avoids code duplication.